<?xml version="1.0"?>
<launch>

  <arg name="camera_name" value="ueye_enac_ceiling_1_6mm" />
  <arg name="camera_id"   value="128" />
  
  <!-- start camera -->
  <!-- -->
  <arg name="nodelet_manager_name" value="cam_nodelet_manager" />
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen" />

  <include file="$(find smocap)/launch/ueye_cam.launch">
    <arg name="camera_name" value="$(arg camera_name)"/>
    <arg name="camera_id" value="$(arg camera_id)"/>
    <arg name="cam_nodelet_id" value="1"/>
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)"/>
    <arg name="camera_parameters_file_alt" value="$(find smocap)/params/f111_cameras_cfg.yaml"/>
  </include>
  
  <!---->

  <!-- spawn marker localization node -->
  <node name="smocap_node" pkg="smocap" type="smocap_node.py">
    <param name="publish_image"   type="bool"   value="true"  />
    <param name="publish_est"     type="bool"   value="true" />
    <param name="cameras"         type="string" value="$(arg camera_name)"/>
    <param name="detect_rgb"      type="bool"   value="false"  />
    <param name="detect_min_area" type="int"    value="12"    />
  </node>

  <!-- spawn static transform publishers
         - world to camera_link
         - camera_link to camera_optical_frame
         - world to map
  -->

  <!--
  <node  name="world_to_cam_optical_frame_publisher" pkg="tf" type="static_transform_publisher"
	 args="1.79896354  1.33057643  2.81374198 -0.70610063  0.70809591  0.00423334 -0.00204123 world $(arg camera_name)_optical_frame 100"/>
  -->

  <!-- again something fishy here: this is cam to world but i send it as world to cam -->
<!--  <node  name="cam_optical_frame_to_world_frame_publisher" pkg="tf" type="static_transform_publisher"
	 args="1.34435324  1.77016933  2.82543134 -0.70610063  0.70809591  0.00423334  0.00204123 world $(arg camera_name)_optical_frame 100"/> -->
   <node  name="cam_optical_frame_to_world_frame_publisher" pkg="tf" type="static_transform_publisher"
         args="1.36602556  1.77145808  2.87497458  0.70733819 -0.70683061 -0.00752713 -0.00255303 world $(arg camera_name)_optical_frame 100"/>
  <!--
   <node  name="cam_optical_frame_to_world_frame_publisher" pkg="tf" type="static_transform_publisher"
   args="1.81135575  1.34714328  2.85897949 0.70733819 -0.70683061 -0.00752713  0.00255303 world $(arg camera_name)_optical_frame 100"/>
   -->
   

  <node  name="cam_optical_frame_to_cam_link_publisher" pkg="tf" type="static_transform_publisher"
	args="0 0 0 0.5 -0.5 0.5 0.5 $(arg camera_name)_optical_frame $(arg camera_name)_link 50"/>

  <!-- -->
  <node  name="world_to_map_publisher" pkg="tf" type="static_transform_publisher"
	args="0 0 0 0 0 0 world map 100"/>
  <!-- -->
	
</launch>

